### ISSUE TITLE
[4.16rc1] Not able to uninstall plugins in Windows

### ISSUE THREAD

**Issue by lauramarcos** [napari/napari#4484]
## 🐛 Bug

When clicking on <Uninstall> button, "uninstall" legend shows up next to plugin name but plugin is not being uninstalled successfully

## To Reproduce

1. Open napari 4.16rc1 in Windows
2. Click on Plugins -> Install/Uninstall Plugins menu
3. In opened modal, click on <Uninstall> button

## Expected behavior

Plugin should show up now in "Available Plugins" section and no longer be listed in Plugins menu

## Environment

napari: 0.4.16rc1
Platform: Windows-10-10.0.17763-SP0
Python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:59:45) [MSC v.1929 64 bit (AMD64)]
Qt: 5.12.9
PySide2: 5.13.2
NumPy: 1.22.3
SciPy: 1.8.0
Dask: 2022.05.0
VisPy: 0.9.6

OpenGL:
- GL version: 4.5.0 NVIDIA 370.41
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1691x989, scale 1.0

Plugins:
- affinder: 0.2.3
- bbii-decon: 0.0.1
- console: 0.0.4
- scikit-image: 0.4.16rc1
- svg: 0.1.5

https://user-images.githubusercontent.com/63799148/167011044-ee62684c-7bbf-4e36-9028-27362b8dc8f8.mov

**Comment by tlambert03** [comment]
does this happen for all plugins?  or is bbii-decon the only one you've found so far?

does the status (bottom left) show any errors?

**Comment by lauramarcos** [comment]
@tlambert03 Yes, it happens with all plugins and no, no error on bottom left but i get the following error:

`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Windows\\gen_py\\3.8\\__init__.py'`

**Comment by tlambert03** [comment]
that's certainly intriguing... is there any more to that trace?

**Comment by lauramarcos** [comment]
@tlambert03 this is the whole error:

```
Traceback (most recent call last):
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\Scripts\mamba-script.py", line 6, in <module>
    from mamba.mamba import main
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\mamba\mamba.py", line 53, in <module>
    from mamba.mamba_shell_init import shell_init
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\mamba\mamba_shell_init.py", line 8, in <module>
    from conda.core.initialize import initialize, initialize_dev, make_initialize_plan
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\conda\core\initialize.py", line 75, in <module>
    from menuinst._legacy.knownfolders import get_folder_path, FOLDERID
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\menuinst\__init__.py", line 4, in <module>
    from .api import *
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\menuinst\api.py", line 10, in <module>
    from .platforms import Menu, MenuItem
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\menuinst\platforms\__init__.py", line 20, in <module>
    Menu, MenuItem = menu_api_for_platform()
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\menuinst\platforms\__init__.py", line 6, in menu_api_for_platform
    from .win import WindowsMenu as Menu, WindowsMenuItem as MenuItem
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\menuinst\platforms\win.py", line 10, in <module>
    from win32com.client import Dispatch
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\__init__.py", line 11, in <module>
    from . import gencache
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\gencache.py", line 761, in <module>
    __init__()
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\gencache.py", line 62, in __init__
    Rebuild()
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\gencache.py", line 746, in Rebuild
    _SaveDicts()
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\gencache.py", line 73, in _SaveDicts
    f = open(os.path.join(GetGeneratePath(), "dicts.dat"), "wb")
  File "D:\Users\lmarcos\AppData\Local\napari-app-0.1\envs\napari-0.4.16rc1\lib\site-packages\win32com\client\gencache.py", line 153, in GetGeneratePath
    f = open(fname, "w")
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Windows\\gen_py\\3.8\\__init__.py'
```

**Comment by goanpeca** [comment]
Will take a look @lauramarcos, thanks for the report!

**Comment by jaimergp** [comment]
This is the same issue reported in #4247, which means that #4462 was not the only fix we needed. I guess there's a separate method for uninstalls I didn't see, because _installing_ plugins does work, right?

**Comment by jaimergp** [comment]
That said, I can only see one method that creates the mamba process... and that one is patched already. Maybe we should pass `--no-shortcuts` to the plugin install commands anyway. I'll see if a PR with those changes fixes it!

**Comment by goanpeca** [comment]
Will try that one. But I checked yesterday with rc1 and plugins where installing 🤔

**Comment by jaimergp** [comment]
I think it might be somehow exclusive to the AWS Workspace configuration. Check the PR I opened on #4520. The artifacts will be ready in the CI output. I can't log in my Workspace now for whatever reason:

<img width="546" alt="image" src="https://user-images.githubusercontent.com/2559438/168126425-9bc7d848-9ec5-4461-a07e-a925b64782aa.png">

**Comment by jaimergp** [comment]
Fix now available in #4520
